src/extent.c

```diff
@@ -502,14 +489,16 @@ static edata_t *
 extent_recycle_split(tsdn_t *tsdn, pac_t *pac, ehooks_t *ehooks,
     ecache_t *ecache, edata_t *expand_edata, size_t size, size_t alignment,
     edata_t *edata, bool growing_retained) {
+	malloc_mutex_assert_owner(tsdn, &ecache->mtx);
```

Comment:
Presumably this verifies that the lock is held, and implies that we should use extent_deactivate_locked below?

Reply:
So one big change in this diff is moving the locking outside this function and locking once rather than repeatedly.